### PR TITLE
Feature/singlepage

### PR DESCRIPTION
**Pull request by schorfES**
Singlepage module with support of transitions. Documentation can be found here: https://github.com/moccu/picnic/tree/feature/singlepage#singlepage

### CODE REVIEW

File: src/singlepage/commands/Navigate.js [hunk 1]
```diff
@@ -5,176 +5,257 @@ import settings from 'picnic/singlepage/settings';
...
+			.then(this._notifyStart.bind(this))
+
+			// Destroy old modules from target...
+			.then(this._destroyOldModules.bind(this))
+
+			// Replace contents in target...
+			.then(this._replaceLoadedContents.bind(this))
+
+			// Update the user's navigation history
+			.then(this._updateHistroy.bind(this))
```
Comment (napanee) [inline src/singlepage/commands/Navigate.js]: typo ;)

File: src/singlepage/commands/Navigate.js [hunk 2]
```diff
@@ -5,176 +5,257 @@ import settings from 'picnic/singlepage/settings';
...
-			if (target && target.length === 1) {
-				top = target.offset().top;
-			}
+	_replaceLoadedContents() {
+		_.each(this.context.getObject(settings.namespaceViews), function(view) {
+			view.replace(this._contents);
+		}, this);
+	}
+
+	_updateHistroy() {
```
Comment (napanee) [inline src/singlepage/commands/Navigate.js]: typo

File: src/singlepage/commands/Navigate.js [hunk 3]
```diff
@@ -5,176 +5,257 @@ import settings from 'picnic/singlepage/settings';
...
+			.then(this._insertScripts.bind(this))
+
+			// Append (reappend) styles
+			.then(this._insertStyles.bind(this))
+
+			// Initialize new modules in target:
+			.then(this._initializeNewModules.bind(this))
+
+			// Inform about a final navigation change...
+			.then(this._notifiyEnd.bind(this))
```
Comment (napanee) [inline src/singlepage/commands/Navigate.js]: typo

File: src/singlepage/commands/Navigate.js [hunk 4]
```diff
@@ -5,176 +5,257 @@ import settings from 'picnic/singlepage/settings';
...
-			this.eventData.href + ' ' + this._settings.selectorUpdate,
-			$.proxy(this._onResponse, this)
-		);
+	_notifyStart() {
+		this.context.dispatch(this.eventName + ':start', $.extend({
+			title: this._title
+		}, this.eventData));
+	}
+
+	_notifiyEnd() {
```
Comment (napanee) [inline src/singlepage/commands/Navigate.js]: typo

File: src/singlepage/commands/Navigate.js [hunk 5]
```diff
@@ -5,176 +5,257 @@ import settings from 'picnic/singlepage/settings';
...
+		this.context.dispatch(this._settings.eventNameInitialize, {
+			root: this._target
+		});
+	}
 
-			if (target && target.length === 1) {
-				top = target.offset().top;
-			}
+	_replaceLoadedContents() {
+		_.each(this.context.getObject(settings.namespaceViews), function(view) {
```
Comment (napanee) [inline src/singlepage/commands/Navigate.js]: use arrow function

**Review by napanee — CHANGES_REQUESTED** [review]

**Comment by schorfES** [comment]
@Napanee the fixes are done. See commits:
- f8703441e66ede2744c0d1982d30cae583a1c9a6
- beb02cf7b82cfe94a5aef016c3083fcd9e1aac97

**Pull request merged**